apifederator/pkg/apis/apifederation/types.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apifederation
+
+import (
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+)
+
+// APIServerList is a list of APIServer objects.
+type APIServerList struct {
+	unversioned.TypeMeta
+	unversioned.ListMeta
+
+	Items []APIServer
+}
+
+// APIServerSpec contains information for locating and communicating with a server.
+// Only https is supported, though you are able to disable certificate verification.
+type APIServerSpec struct {
+	// InternalHost is the host:port for locating the server inside the pod/service network.
+	InternalHost string
+	// Group is the API group name this server hosts
+	Group string
+	// Version is the API version this server hosts
+	Version string
```

Comment:
> (I was actually expecting to split only by group. I guess it does seem valuable to host different versions separately in some situations, but is it really necessary?)

This came out of my first encounter with the enemy as I was dealing with an openshift project API that had one version hosted in openshift and another hosted in a third server that works against bare kube.  It got me thinking about what I wanted to do in an upgrade scenario and I realized I wanted to avoid risking my v1 server.